### PR TITLE
add one more test to catch a potential bug

### PR DESCRIPTION
**Pull request by stevemao**

### CODE REVIEW

File: test/Course/StateTest.hs [hunk 1]
```diff
@@ -94,6 +94,8 @@ monadTest =
   testGroup "Monad" [
     testCase "(=<<)" $
       runState (const (put 2) =<< put 1) 0 @?= ((),2)
+  , testCase "" $
```
Comment (gwils) [inline test/Course/StateTest.hs]: Please give this test a name rather than an empty string.

**Comment by gwils** [comment]
Hi, thanks for submitting this.
Could you please explain what bug this test prevents that the other tests don't catch?

**Comment by stevemao** [comment]
My code:

```hs
  f =<< a = State b
    where b s = (eval state s, exec state . exec a $ s)
                  where state = f . eval a $ s
```

existing tests doest catch this but will fail later on.

**Comment by gwils** [comment]
Thank you, this test is helpful.

**Pull request merged**